### PR TITLE
Translate wikis to Chinese

### PR DESCRIPTION
**Pull request by Tolu-Mals**
# Translate Wikis to Chinese
Adds support for chinese wiki translations using translate button

## How should this be tested?
1. Switch language to chinese using language selector in navbar
2. Open a wiki page
3. Click the translate button ( button with 'ZH' label)

## Demo
https://github.com/EveripediaNetwork/ep-ui/assets/30352484/5f9b07be-8720-4801-8021-976347ccbaad

## Linked issues
closes https://github.com/EveripediaNetwork/issues/issues/2376

### CODE REVIEW

**Review by Simplypheyie96 — APPROVED** [review]

**Review by Adebesin-Cell — COMMENTED** [review]
Screenshot 2024-03-13 at 4 31 32 PM

Screenshot 2024-03-13 at 4 31 26 PM

It seems to remove some markups after translating, like the strikethrough was removed

**Comment by Tolu-Mals** [comment]
> Screenshot 2024-03-13 at 4 31 32 PM
> 
> Screenshot 2024-03-13 at 4 31 26 PM
> 
> It seems to remove some markups after translating, like the strikethrough was removed

I'll see if I can fix that through prompting.

**Comment by Simplypheyie96** [comment]
> Is their any reason why the translator switch is not showing by default until use the switch in the nav @Tolu-Mals

The switch is only for other languages and not for english since english is the default language.

**Comment by Tolu-Mals** [comment]
> Is their any reason why the translator switch is not showing by default until use the switch in the nav @Tolu-Mals

Yeah that was part of the requirements for the translation button

**Comment by Tolu-Mals** [comment]
> Screenshot 2024-03-13 at 4 31 32 PM
> 
> Screenshot 2024-03-13 at 4 31 26 PM
> 
> It seems to remove some markups after translating, like the strikethrough was removed

I'm not sure I can fix this, translations aren't usually verbatim, plus LLMs aren't so predictable

File: src/pages/api/translate-wiki.ts [hunk 1]
```diff
@@ -20,22 +29,9 @@ export default async function handler(
         messages: [
           {
             role: 'system',
-            content: `A peice of markdown will be provided after the keyword 'START'. Translate the peiece of markdown text to Korean, using the following rules:
-  1. Do not translate names of people or organizations to Korean
+            content: `A peice of markdown will be provided after the keyword 'START'. Translate the peiece of markdown text to ${targetLanguage}, using the following rules:
```
Comment (Adebesin-Cell) [inline src/pages/api/translate-wiki.ts:32]: ``piece``

File: src/pages/api/translate-wiki.ts [hunk 2]
```diff
@@ -20,22 +29,9 @@ export default async function handler(
...
+  1. Do not translate names of people or organizations to ${targetLanguage}
   2. Any peice of characters that start and end with double asterisks(**) should be immediately fellowed by an empty space( ).
```
Comment (Adebesin-Cell) [inline src/pages/api/translate-wiki.ts:34]: ``piece``

**Review by Adebesin-Cell — COMMENTED** [review]
Screenshot 2024-03-18 at 12 58 35 PM
I searched frax finance after translating IQ.wiki, when I tried translating it was giving for IQ.wiki not frax.

**Comment by Tolu-Mals** [comment]
> Screenshot 2024-03-18 at 12 58 35 PM I searched frax finance after translating IQ.wiki, when I tried translating it was giving for IQ.wiki not frax.

I don't think that's related to translation, seems like the state for wiki wasn't updated in the component.  We can have another issue for that.

**Comment by Adebesin-Cell** [comment]
> > Screenshot 2024-03-18 at 12 58 35 PM I searched frax finance after translating IQ.wiki, when I tried translating it was giving for IQ.wiki not frax.
> 
> I don't think that's related to translation, seems like the state for wiki wasn't updated in the component. We can have another issue for that.

Oh, okay.

**Review by Adebesin-Cell — APPROVED** [review]

**Pull request merged**